### PR TITLE
AzCopy command reference text updated

### PR DESCRIPTION
**Pull request by normesta**
Recommended updates to AzCopy command help reference text. This includes descriptions in the helpMessages.go file as well as parameter descriptions in various command go files.

### CODE REVIEW

File: cmd/helpMessages.go [hunk 1]
```diff
@@ -3,283 +3,435 @@ package cmd
...
 const copyCmdShortDescription = "Copies source data to a destination location"
 
 const copyCmdLongDescription = `
-Copies source data to a destination location. The supported directions are:
-  - local <-> Azure Blob (SAS or OAuth authentication)
-  - local <-> Azure File (Share/directory SAS authentication)
-  - local <-> ADLS Gen 2 (SAS, OAuth, or SharedKey authentication)
-  - Azure Blob (SAS or public) <-> Azure Blob (SAS or OAuth authentication)
-  - Azure File (SAS) -> Azure Block Blob (SAS or OAuth authentication)
-  - AWS S3 (Access Key) -> Azure Block Blob (SAS or OAuth authentication)
```
Comment (zezha-msft) [inline cmd/helpMessages.go]: why is this removed?

File: cmd/copy.go [hunk 2]
```diff
@@ -1148,34 +1148,33 @@ func init() {
...
+	cpCmd.PersistentFlags().Float64Var(&raw.blockSizeMB, "block-size-mb", 0, "Use this block size (specified in MiB) when uploading to Azure Storage, and downloading from Azure Storage. The default value is automatically calculated based on file size. Decimal fractions are allowed (For example: 0.25).")
+	cpCmd.PersistentFlags().StringVar(&raw.logVerbosity, "log-level", "INFO", "Define the log verbosity for the log file, available levels: INFO(all requests/responses), WARNING(slow responses), ERROR(only failed requests), and NONE(no output logs). (default 'INFO').")
+	cpCmd.PersistentFlags().StringVar(&raw.blobType, "blob-type", "None", "Defines the type of blob at the destination. This is used for uploading blobs and when copying between accounts (default 'None').")
+	cpCmd.PersistentFlags().StringVar(&raw.blockBlobTier, "block-blob-tier", "None", "Upload block blob to Azure Storage using this blob tier. (default 'None').")
+	cpCmd.PersistentFlags().StringVar(&raw.pageBlobTier, "page-blob-tier", "None", "Upload page blob to Azure Storage using this blob tier. (default "None").")
```
Comment (zezha-msft) [inline cmd/copy.go]: >(default "None") [](start = 137, length = 16)

The quotes need to be escaped with \.

File: cmd/helpMessages.go [hunk 3]
```diff
@@ -3,283 +3,435 @@ package cmd
 import "github.com/Azure/azure-storage-azcopy/common"
 
 // ===================================== ROOT COMMAND ===================================== //
-const rootCmdShortDescription = "AzCopy is a command line tool that moves data into/out of Azure Storage."
+const rootCmdShortDescription = "AzCopy is a command-line tool that moves data into and out of Azure Storage."
 
 const rootCmdLongDescription = "AzCopy " + common.AzcopyVersion +
 	`
-Project URL: github.com/Azure/azure-storage-azcopy
+  The general format of the commands is: ``azcopy [command] [arguments] --[flag-name]=[flag-value]``.
```
Comment (zezha-msft) [inline cmd/helpMessages.go]: >`` [](start = 41, length = 2)

this file doesn't compile any more.

Please try installing Go here ([https://golang.org/](https://golang.org/)) and compile the tool using: go build -o azcopy.exe, and make sure the help messages are valid Go strings

File: cmd/helpMessages.go (hunk 1 above)
Comment (normesta) [reply cmd/helpMessages.go]: Good catch. Ok I added the text back.

File: cmd/copy.go (hunk 2 above)
Comment (normesta) [reply cmd/copy.go]: Ok. I fixed all errors and compiled successfully. I had to add some new constants to support the formatting of code blocks in the output.

File: cmd/helpMessages.go (hunk 3 above)
Comment (normesta) [reply cmd/helpMessages.go]: Ok. I fixed all errors and compiled successfully. I had to add some new constants to support the formatting of code blocks in the output.

File: cmd/env.go [hunk 4]
```diff
@@ -43,6 +43,6 @@ var envCmd = &cobra.Command{
 }
 
 func init() {
-	envCmd.PersistentFlags().BoolVar(&showSensitive, "show-sensitive", false, "Show sensitive/secret environment variables")
+	envCmd.PersistentFlags().BoolVar(&showSensitive, "show-sensitive", false, "Shows sensitive/secret environment variables")
```
Comment (adreed-msft) [inline cmd/env.go]: Admittedly I'm the one to blame for the source text here, but consider changing this to "Displays the value of sensitive/secret environment variables" instead. I feel like the definition of this is unclear until used, in hindsight.

File: cmd/remove.go [hunk 5]
```diff
@@ -82,8 +82,8 @@ func init() {
...
-	deleteCmd.PersistentFlags().StringVar(&raw.include, "include", "", "only include files whose name matches the pattern list. Example: *.jpg;*.pdf;exactName")
-	deleteCmd.PersistentFlags().StringVar(&raw.exclude, "exclude", "", "exclude files whose name matches the pattern list. Example: *.jpg;*.pdf;exactName")
+	deleteCmd.PersistentFlags().BoolVar(&raw.recursive, "recursive", false, "Look into sub-directories recursively when syncing between directories.")
+	deleteCmd.PersistentFlags().StringVar(&raw.logVerbosity, "log-level", "INFO", "Define the log verbosity for the log file. Available levels include: INFO(all requests/responses), WARNING(slow responses), ERROR(only failed requests), and NONE(no output logs). (default 'INFO')")
+	deleteCmd.PersistentFlags().StringVar(&raw.include, "include", "", "Only include files where the name matches the pattern list. For example: .jpg;.pdf;exactName")
+	deleteCmd.PersistentFlags().StringVar(&raw.exclude, "exclude", "", "Exclude files where the name matches the pattern list. For example: .jpg;.pdf;exactName")
```
Comment (adreed-msft) [inline cmd/remove.go]: >.jpg;.pdf;exactName [](start = 137, length = 19)

In our include and exclude filters, we make use of path.Match. This means that we'll listen to wildcards, and that these would just match to files exactly named .jpg, .pdf, and exactName.

Please keep the wildcards, otherwise this is invalid documentation. If this is formatting as markdown, manual edits may have to be made when generated.

File: cmd/sync.go [hunk 6]
```diff
@@ -507,15 +507,14 @@ func init() {
...
-	syncCmd.PersistentFlags().BoolVar(&raw.putMd5, "put-md5", false, "create an MD5 hash of each file, and save the hash as the Content-MD5 property of the destination blob/file. (By default the hash is NOT created.) Only available when uploading.")
-	syncCmd.PersistentFlags().StringVar(&raw.md5ValidationOption, "check-md5", common.DefaultHashValidationOption.String(), "specifies how strictly MD5 hashes should be validated when downloading. Only available when downloading. Available options: NoCheck, LogOnly, FailIfDifferent, FailIfDifferentOrMissing.")
+	syncCmd.PersistentFlags().BoolVar(&raw.recursive, "recursive", true, "True by default, look into sub-directories recursively when syncing between directories. (default true).")
+	syncCmd.PersistentFlags().Float64Var(&raw.blockSizeMB, "block-size-mb", 0, "Use this block size (specified in MiB) when uploading to Azure Storage or downloading from Azure Storage. Default is automatically calculated based on file size. Decimal fractions are allowed (For example: 0.25).")
+	syncCmd.PersistentFlags().StringVar(&raw.include, "include", "", "Include only files where the name matches the pattern list. For example: .jpg;.pdf;exactName.")
```
Comment (adreed-msft) [inline cmd/sync.go]: ditto on wildcards.

**Comment by JohnRusk** [comment]
Norm, we'll need to figure out what to do about the * issue noted above.  Marking as "on hold" until we figure that out.  I think Ze will touch base with you.

**Comment by JohnRusk** [comment]
This is the old, obsoelete PR isn't it. (Replaced by a new one?).

**Comment by normesta** [comment]
Thanks @JohnRusk. Yes this is old. This PR was to master. The latest PR from branch "normesta-ref-strings" was to the dev branch. Those changes became live when the dev branch merged with master.  I think I can close this one. Sorry for the confusion. I guess I never close it.
